packages/block-serialization-default-parser/src/index.js

```diff
@@ -162,42 +230,47 @@ export const parse = ( doc ) => {
 	return output;
 };
 
+/**
+ * Parses the next token in the input document.
+ *
+ * @return {boolean} Returns true when there is more tokens to parse.
+ */
 function proceed() {
+	const stackDepth = stack.length;
 	const next = nextToken();
+	if ( next === null ) {
```

Comment:
This refactoring was driven by TypeScript that wasn't quite happy about the token being an array with 1 param or 5 params. It turns out that when using `null` the behavior remains the same and we can bail out earlier and avoid other operations that aren't necessary for this case anyway.

Reply:
did you look at using `[TokenType, string?, Attributes?, number?, number?]`?

the refactor looks fine but I'd rather we return `'no-more-tokens'` than `null` because of the self-documenting nature of the string value.

Reply:
Yes, that was my initial try, but TS was complaining about the number of items in the array returned: 1 instead of 5. Maybe `return [ 'no-more-tokens', , , , ];` would work then?

Reply:
I opened #44459 to address it.